server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -252,8 +252,14 @@ public void run()
           try {
             try {
               inputRow = firehose.nextRow();
+
+              if (inputRow == null) {
+                log.debug("thrown away null input row, considering unparseable");
```

Comment:
I definitely think validation pre-Druid is the way to go, so I agree with the statement that unparseable events should be rare and probably indicate something wrong with the data pipeline.